/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(head == null || head.next == null){
            return head;
        }
        int x = 1;
        ListNode tail = head;
        while(tail.next != null){
            tail = tail.next;
            x++;
        }
        k = k % x;
        if(k == 0){
            return head;
        }
        tail.next = head;
        for(int i = 0; i < x - k; i++){
            tail = tail.next;
        }
        head = tail.next;
        tail.next = null;

        return head;
    }
}